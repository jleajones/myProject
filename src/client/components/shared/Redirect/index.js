import React from 'react';
import { Route, Switch, Redirect as Original } from 'react-router-dom';
import PropTypes from 'prop-types';

const Redirect = ({ from, to }) => (
  <Route
    render={({ staticContext }) => {
      if (staticContext) {
        staticContext.statusCode = 302;
      }

      return Array.isArray(from) ? (
        <Switch>
          {from.map(cf => (
            <Original key={cf} from={cf} to={to} />
          ))}
        </Switch>
      ) : (
        <Original key={from} from={from} to={to} />
      );
    }}
  />
);

Redirect.propTypes = {
  to: PropTypes.string.isRequired,
  from: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Redirect;
