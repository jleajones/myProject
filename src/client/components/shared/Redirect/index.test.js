import React from 'react';
import { shallow } from 'enzyme';

import Redirect from '.';

describe('<Redirect />', () => {
  test('Renders component without crashing', () => {
    const component = shallow(
      <Redirect from={['/new-home-page', '/home-page-2']} to="/" />
    );

    expect(component).toMatchSnapshot();
  });
});
