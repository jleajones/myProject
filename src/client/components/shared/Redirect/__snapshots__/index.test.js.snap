// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Redirect /> Renders component without crashing 1`] = `
<Route
  render={[Function]}
/>
`;
